ambry-messageformat/src/test/java/com.github.ambry.messageformat/BlobPropertiesTest.java

```diff
@@ -63,8 +101,82 @@ public void basicTest() {
+      blobProperties =
+          getBlobProperties(blobSize, serviceId, ownerId, contentType, true, ttl, creationTimeMs, accountId,
+              containerId, creatorAccountId, version);
+      verifyBlobProperties(blobProperties, blobSize, serviceId, ownerId, contentType, true, ttl, accountId, containerId,
+          creatorAccountId);
+    }
+  }
+
+  /**
+   * Creates {@link BlobProperties} based on the args passed for the given version
+   * @param blobSize the size of the blob
+   * @param serviceId the serviceId associated with the {@link BlobProperties}
+   * @param ownerId the ownerId associated with the {@link BlobProperties}
+   * @param contentType the contentType associated with the {@link BlobProperties}
+   * @param isPrivate refers to whether the blob is private or not
+   * @param timeToLiveInSeconds the time to live associated with the {@link BlobProperties} in secs
+   * @param creationTimeMs creation time of the blob in ms
+   * @param accountId accountId of the user who uploaded the blob
+   * @param containerId containerId of the blob
+   * @param creatorAccountId Issuer AccountId of the put request
+   * @param version the version in which {@link BlobProperties} needs to be created
+   * @return the {@link BlobProperties} thus created
+   */
+  private BlobProperties getBlobProperties(long blobSize, String serviceId, String ownerId, String contentType,
+      boolean isPrivate, long timeToLiveInSeconds, long creationTimeMs, short accountId, short containerId,
+      short creatorAccountId, short version) {
+    if (version == BlobPropertiesSerDe.Version1) {
+      return new BlobProperties(blobSize, serviceId, ownerId, contentType, isPrivate, timeToLiveInSeconds,
+          creationTimeMs);
+    } else {
+      return new BlobProperties(blobSize, serviceId, ownerId, contentType, isPrivate, timeToLiveInSeconds,
+          creationTimeMs, accountId, containerId, creatorAccountId);
+    }
+  }
+
+  /**
+   * Creates {@link BlobProperties} based on the args passed for the given version
+   * @param blobSize the size of the blob
+   * @param serviceId the serviceId associated with the {@link BlobProperties}
+   * @param ownerId the ownerId associated with the {@link BlobProperties}
+   * @param contentType the contentType associated with the {@link BlobProperties}
+   * @param isPrivate refers to whether the blob is private or not
+   * @param timeToLiveInSeconds the time to live associated with the {@link BlobProperties} in secs
+   * @param accountId accountId of the user who uploaded the blob
+   * @param containerId containerId of the blob
+   * @param creatorAccountId Issuer AccountId of the put request
+   * @param version the version in which {@link BlobProperties} needs to be created
+   * @return the {@link BlobProperties} thus created
+   */
+  private BlobProperties getBlobProperties(long blobSize, String serviceId, String ownerId, String contentType,
```

Comment:
each of them tests different constructors of BlobProperties and there are prod use-cases for each of these constructors and hence.